commands/post.go

```diff
@@ -125,11 +132,30 @@ func printPost(c client.Client, post *model.Post, usernames map[string]string, s
 		}
 	}
 
-	if showIds {
-		printer.PrintT(fmt.Sprintf("\u001b[31m%s\u001b[0m \u001b[34;1m[%s]\u001b[0m {{.Message}}", post.Id, username), post)
+	postTime := model.GetTimeForMillis(post.CreateAt)
+	createdAt := postTime.Format(PostTimeFormat)
+
+	if showTimestamp {
+		printer.PrintT(fmt.Sprintf("\u001b[32m%s\u001b[0m \u001b[34;1m[%s]\u001b[0m {{.Message}}", createdAt, username), post)
 	} else {
-		printer.PrintT(fmt.Sprintf("\u001b[34;1m[%s]\u001b[0m {{.Message}}", username), post)
+		if showIds {
+			printer.PrintT(fmt.Sprintf("\u001b[31m%s\u001b[0m \u001b[34;1m[%s]\u001b[0m {{.Message}}", post.Id, username), post)
+		} else {
+			printer.PrintT(fmt.Sprintf("\u001b[34;1m[%s]\u001b[0m {{.Message}}", username), post)
+		}
+	}
+}
+
+func getPostList(client client.Client, channelID, since string, perPage int) (*model.PostList, *model.Response, error) {
+	if len(since) > 0 {
+		sinceTime, err := time.Parse(ISO8601Layout, since)
+		if err != nil {
+			return nil, nil, fmt.Errorf("invalid since time format: %v", err)
+		}
+		sinceTimeMillis := model.GetMillisForTime(sinceTime)
+		return client.GetPostsSince(channelID, sinceTimeMillis, false)
 	}
+	return client.GetPostsForChannel(channelID, 0, perPage, "", false)
```

Comment:
We can save some indentation and it's more idiomatic to check if the string is empty rather than `len` check.
```suggestion
	if since == "" {
		return client.GetPostsForChannel(channelID, 0, perPage, "", false)
	}

	sinceTime, err := time.Parse(ISO8601Layout, since)
	if err != nil {
		return nil, nil, fmt.Errorf("invalid time format for since flag: %v", err)
	}

	sinceTimeMillis := model.GetMillisForTime(sinceTime)
	return client.GetPostsSince(channelID, sinceTimeMillis, false)
```

Reply:
Suggestion committed